initialized working branch

SUBstylee opened the pull request:
will be pushing changes from here

SUBstylee committed:
initialized working branch
will be pushing changes from here

SUBstylee committed:
added test
added smoke and snapshot testing for components.
this will be an ongoing effort.

ryantang333 commented:
One note - typically you don't have to open your PR before your work is done, but it's also nice to see progress.

ryantang333 commented:
Also, what feature is this branch for?

SUBstylee committed:
brought in redux
removed created a store and now using redux
also added components: loader and message
